### ISSUE TITLE
[Bug]: Authentication failure

### ISSUE THREAD

**Issue by jdeluyck** [bvis/aegis-hass#526]
### Describe the bug

I have a plain Ajax system, and created a new user + set the password (I can authenticate an Ajax app with username+pw+TOTP)

Attempting to log into the user with the integration always returns "Invalid username or password". There's nothing useful in the logs. I've tried logging in from multiple devices (all android smartphones), and user+pw+TOTP always works.

### Steps to reproduce

Try to log in

### Expected behavior

It to log in

### Home Assistant version

2026.9.3

### Integration version

1.22.0

### Installation method

HACS

### App label (co-branded app)

Ajax (plain, not rebranded)

### FCM push notifications configured?

No

### Relevant logs

```text
Logger: custom_components.aegis_ajax.config_flow
Source: custom_components/aegis_ajax/config_flow.py:195
Integration: Aegis for Ajax (documentation, issues)
First occurred: 12:06:07 (1 occurrence)
Last logged: 12:06:07
Authentication failed: Invalid email or password
```

### Screenshots

<img width="554" height="537" alt="Image" src="https://github.com/user-attachments/assets/b69697a7-e7c4-4467-b1d6-6c1a38b4907e" />

**Comment by bvis** [comment]
Thanks for the report. "Invalid email or password" is the answer of Ajax's own login endpoint, not something the integration decides — and it is returned *before* two-factor comes into play: an account with 2FA gets a code step after a correct email + password, so here Ajax rejected the pair itself. Debug logging will not add anything; the server sends only that rejection.

That leaves the pair itself, and three things routinely produce this while the app keeps working:

1. **The app is not sending a password.** Log **out** of the Ajax app on one phone and log back **in typing the email and password by hand** — not Google/Apple sign-in, not the saved-password autofill. If that fails, the account has no password (or not the one you think) and setting it from the app's password-recovery flow fixes both.
2. **Characters outside plain ASCII, or a leading/trailing space.** Passwords are hashed exactly as typed. As a test, set a temporary password using only letters and digits and try Home Assistant with it.
3. **Email spelling.** Copy the address exactly as the app shows it under the account profile (case included) rather than retyping it.

If all three check out and it still fails, say which Ajax app the account signs in with (the regular Ajax Security System app or the PRO app) and I will take it from there.

**Comment by jdeluyck** [comment]
Is there a limit on the password length? Mine is rather long 😅

**Comment by bvis** [comment]
No length limit, neither here nor on the wire. The integration never sends your password: it sends its SHA-256 digest, which is 64 hex characters whatever you typed — a 12-character password and a 60-character one produce requests of identical size. So length is not what Ajax is rejecting.

What does matter is that the bytes are exactly the ones the app hashes, which is why the three checks above are the ones to run — in particular a trailing space or newline picked up when pasting from a password manager, or a character outside plain ASCII. The quickest discriminator is still point 1: log out of the Ajax app and log back in typing the password by hand.
